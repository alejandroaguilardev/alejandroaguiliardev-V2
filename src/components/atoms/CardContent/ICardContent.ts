export interface ICardContent {
	title: string;
	subtitle: string;
	description: string;
	icon: string;
	size?:string;
}
