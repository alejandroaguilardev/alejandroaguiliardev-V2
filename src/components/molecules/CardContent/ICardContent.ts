export interface ICardContent {
	title: string;
	subtitle: string;
	description: string;
	icon: string;
	size?:string;
	link?:string;
	repository?:string;
}
