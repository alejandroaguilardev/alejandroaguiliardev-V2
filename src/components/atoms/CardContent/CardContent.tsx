import Image from "next/image"
import { Box, Typography } from "@mui/material"
import { ICardContent } from './ICardContent'

export const CardContent = ({ title, subtitle, description, icon, size = 'small' }: ICardContent) => {
    return (
        <>
            <Box sx={{ py: 1 }}>
                <Image
                    src={icon}
                    width={size === "small" ? 70 : 100}
                    height={size === "small" ? 70 : 100}
                    alt={title}
                />
            </Box>
            <Typography align='center' variant='h3' sx={{ fontSize: "1rem", fontWeight: 'bold', mb: 1 }}>
                {title}
            </Typography>
            <Typography align='center' sx={{ fontSize: ".9rem" }}>
                {subtitle}
            </Typography>

            <Typography align='center' sx={{ fontSize: ".7rem" }}>
                {description}
            </Typography>
        </>
    )
}
