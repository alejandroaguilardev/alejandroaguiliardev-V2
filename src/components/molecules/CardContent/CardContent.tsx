import { Box, Typography } from "@mui/material"
import { CardImage } from "../CardImage/CardImage"
import { ICardContent } from './ICardContent'

export const CardContent = ({
    title,
    subtitle,
    description,
    icon,
    size = 'small',
    link = "",
    repository = ""
}: ICardContent) => {
    return (
        <>
            <CardImage
                title={title}
                icon={icon}
                size={size}
                link={link}
                repository={repository}
            />
            <Box px={2}>
                <Typography align='center' variant='h3' sx={{ fontSize: "1rem", fontWeight: 'bold', mb: 1 }}>
                    {title}
                </Typography>
                <Typography align='center' sx={{ fontSize: ".9rem" }}>
                    {subtitle}
                </Typography>

                <Typography align='center' sx={{ fontSize: ".7rem" }}>
                    {description}
                </Typography>
            </Box>
        </>
    )
}
