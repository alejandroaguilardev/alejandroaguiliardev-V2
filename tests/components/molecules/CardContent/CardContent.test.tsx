import { render, screen } from "@testing-library/react";
import { CardContent } from "../../../../src/components/molecules/CardContent/CardContent";



describe("CardContent Test", () => {
	const title = "title Test";
	const subtitle = "subtitle Test";
	const description = "description Test";
	const icon = "/img/banner.jpg";
	const size = "large";
	const link = "link";
	const repository = "#";

	test("toMatchSnapshot", () => {
		const { container } = render(
			<CardContent
				title={title}
				subtitle={subtitle}
				description={description}
				icon={icon}
			/>
		);
		expect(container).toMatchSnapshot();
	});

	test("Props", () => {
		render(
			<CardContent
				title={title}
				subtitle={subtitle}
				description={description}
				icon={icon}
				size={size}
				link={link}
				repository={repository}
			/> 
		);
		screen.debug()
		expect(screen.findByText(title)).toBeTruthy();
		expect(screen.findByText(subtitle)).toBeTruthy();
		expect(screen.findByText(description)).toBeTruthy();
	});
});
