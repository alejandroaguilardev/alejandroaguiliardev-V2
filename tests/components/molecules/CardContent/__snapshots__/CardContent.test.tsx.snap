// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContent Test Props 1`] = `
<div>
  <div
    class="MuiBox-root css-18i215d"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2770%27%20height=%2770%27/%3e"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="title Test"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
  </div>
  <div
    class="MuiBox-root css-gajhq5"
  >
    <h3
      class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter css-yg6atj-MuiTypography-root"
    >
      title Test
    </h3>
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1a3kej9-MuiTypography-root"
    >
      subtitle Test
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-r5yzcc-MuiTypography-root"
    >
      description Test
    </p>
  </div>
</div>
`;

exports[`CardContent Test toMatchSnapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-18i215d"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2770%27%20height=%2770%27/%3e"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="title Test"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
  </div>
  <div
    class="MuiBox-root css-gajhq5"
  >
    <h3
      class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter css-yg6atj-MuiTypography-root"
    >
      title Test
    </h3>
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1a3kej9-MuiTypography-root"
    >
      subtitle Test
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-r5yzcc-MuiTypography-root"
    >
      description Test
    </p>
  </div>
</div>
`;
